2-planning/0-introduction.adoc

```diff
@@ -0,0 +1,20 @@
+When I was younger and dumber, the planning stage for APIs was pretty much "think of a list of endpoints then start coding" which is wildly insufficient. As the <<theory-contracts,API Contracts>> chapter explained, contracts are incredibly important throughout the life-cycle of an API. We are going to look into how they can be used in the planning and design stage.
+
+Most of the planning and design phase is intended to do two things:
+
+1. Make developers think really hard about the requirements for their API
+2. Gather feedback from clients application developers
+
+Poorly planned APIs waste a lot of time and money for a lot of people. I've seen APIs go from v1 to v2 in the first few months after the clients noticed it didn't actually work properly.
```

Comment:
Maybe add a little bit about how money is wasted. Its clear for people who have done this a lot, but I think for people new to the game, it could be useful.

Reply:
Done! Thank you.